airflow/www/static/js/api/useGridData.ts

```diff
@@ -105,6 +105,7 @@ const useGridData = () => {
       };
       const response = await axios.get<AxiosResponse, GridData>(gridDataUrl, {
         params,
+        paramsSerializer: { indexes: null },
```

Comment:
Nice, I think this should be configured globally at the axios instance level, not per request.

This way if another call also utilise arrays as query param values, it will get serialised in a consistent way.

Reply:
Sure - seems possible here: https://github.com/axios/axios/issues/5217#issuecomment-1379878665

However, do you know if any other requests use the empty bracket behavior currently?

Reply:
Not to my knowledge. We might want to double check that by a quicksearch in our `useQuery` custom hooks but I think we are fine.

Reply:
Not seeing anything obvious but if you could take a second look, that'd be great: https://gist.github.com/vchiapaikeo/c25e688a7cd21e100544807df8c2d3f5

Reply:
The extract is missing some axios calls, for instance in the useClearTaskDryRun, it is cut off too soon, we don’t have the axios call I believe.

But that would be the idea checking in those files if an array is used as query param. Don’t think so

Reply:
The extract is missing some axios calls, for instance in the useClearTaskDryRun, it is cut off too soon, we don’t have the axios call I believe.

But that would be the idea checking in those files if an array is used as query param. Don’t think so

Reply:
[useMarkTaskDryRun](https://github.com/apache/airflow/blob/main/airflow/www/static/js/api/useMarkTaskDryRun.ts#L81-L83) looked a little suspicious to me (useClearTaskDryRun also has a similar forEach loop). However, in the UI, I couldn't find a request that would return multiple map_index. I also did some manual UI testing and watched network calls and couldn't find any array-like ones. I think we are good.

<img width="1470" alt="image" src="https://github.com/apache/airflow/assets/9200263/16ba1644-4615-41d5-8e44-ea2ae9d2a5c6">